src/components/study-container.js

```diff
@@ -356,22 +358,45 @@ export function StudyContainer({ view, onChangeTab }) {
     const fetchStudyPath = useCallback(() => {
         fetchPath(studyUuid)
             .then((response) => {
-                const study = response[0];
                 const parents = response
                     .slice(1)
                     .map((parent) => parent.elementName);
 
+                const path = computeFullPath(parents);
+                if (studyMetadata.path) {
+                    if (studyMetadata.path !== path) {
+                        snackInfo('', 'moveStudyNotification', {
+                            oldStudyPath: studyMetadata.path,
+                            studyPath: path,
+                        });
+                    }
+                }
+
+                const studyName = response[0]?.elementName;
+                if (studyMetadata.name) {
+                    if (studyMetadata.name !== studyName) {
+                        snackInfo('', 'renameStudyNotification', {
+                            oldStudyName: studyMetadata.name,
+                            studyName: studyName,
+                        });
+                    }
+                }
+
                 document.title = computePageTitle(
                     initialTitle,
-                    study?.elementName,
+                    studyName,
                     parents
                 );
+
+                studyMetadata.path = path;
+                studyMetadata.name = studyName;
+                setStudyMetadata(studyMetadata);
```

Comment:
is it safe to call setStudyMetadata here whereas the useCallBack has studyMetadata in dependency array ?

Reply:
Upon research it wasn't best practice compliant so I've overhauled the process through a custom hook